packages/core/src/tile/types/index.ts

```diff
@@ -168,3 +155,54 @@ export const types = {
     }
   }
 };
+
+export const enum RequestType {
+  CANCEL = 0,
+  IMAGE = 1,
+
+  INITIALIZE = 100
+}
+
+export interface DataRequest {
+  type: RequestType;
+  id: string;
+  request: any;
+}
+
+export interface InitializeMessage {
+  token: string;
+  basePath?: string;
+}
+
+export interface ImageMessage {
+  index: number[];
+  tileSize: number;
+  levelRecord: LevelRecord;
+  namespace: string;
+  channelRanges: number[];
+  channelMapping: number[];
+  attribute: Attribute;
+  token: string;
+  basePath: string;
+  dimensions: Dimension[];
+}
+
+export interface WorkerResponse {
+  type: RequestType;
+  id: string;
+  response: any;
+}
+
+export interface ImageResponse {
+  index: number[];
+  data: TypedArray;
+  width: number;
+  height: number;
+  channels: number;
+  dtype: string;
```

Comment:
NIT: As far as i understand dtype is same as the keys of the `types`, you can do the following if that's the case

```suggestion
  dtype: keyof typeof types;
```